mara_db/views.py

```diff
@@ -23,11 +23,44 @@ def navigation_entry():
                 description=f'The schema of the {alias} db',
                 uri_fn=lambda current_db=alias: flask.url_for('mara_db.index_page', db_alias=current_db))
             for alias, db in config.databases().items()
-            if (isinstance(db, dbs.PostgreSQLDB) and not isinstance(db, dbs.RedshiftDB))
-               or isinstance(db, dbs.MysqlDB)  # for now, only show postgres and mysql schemas
+            if supports_extract_schema(db)
         ])
 
 
+@singledispatch
+def supports_extract_schema(db: object) -> [bool]:
+    """
+    Returns true when the db supports schema extraction
+
+    Args:
+        db: The database which shall be tested for schema extraction
+    """
+    return False
+
+
+@supports_extract_schema.register(str)
+def __(alias: str):
+    return supports_extract_schema(dbs.db(alias))
+
+
+@supports_extract_schema.register(dbs.PostgreSQLDB)
+def __(db: dbs.PostgreSQLDB):
+    return not isinstance(db, dbs.RedshiftDB)
```

Comment:
Isn't it easer simply to add another implementation for RedshiftDB which returns False? Shouldn't really matter...

Reply:
hmm ok, maybe the guy doing the RedshiftDB implementation can add the new method. I think we should skip this to avoid blowing up the source code for things which are not supported.